### DISCUSSION TITLE
DEFI section:question about `redeemCollateral` function

### DISCUSSION THREAD

**Discussion by nikita-biichuk** [Cyfrin/foundry-full-course-cu#3053] [General]
I noticed this in the redeemCollateral function:

```
/*
     * @param tokenCollateralAddress: The ERC20 token address of the collateral you're redeeming
     * @param amountCollateral: The amount of collateral you're redeeming
     * @notice This function will redeem your collateral.
     * @notice If you have DSC minted, you will not be able to redeem until you burn your DSC
     */
    function redeemCollateral(
        address tokenCollateralAddress,
        uint256 amountCollateral
    )
        external
        moreThanZero(amountCollateral)
        nonReentrant
        isAllowedToken(tokenCollateralAddress)
    {
        _redeemCollateral(tokenCollateralAddress, amountCollateral, msg.sender, msg.sender);
        revertIfHealthFactorIsBroken(msg.sender);
    }
```
    
    
    
 I see the comment mentions:

"If you have DSC minted, you will not be able to redeem until you burn your DSC."

However, I don’t see any explicit check in the code to ensure the user has burned their DSC before redeeming their collateral. Does this mean the comment is incorrect, and the user can redeem their collateral as long as their health factor is not broken after the transaction? Could someone clarify this?

**Comment by EngrPips** [comment]
Hello @nikitabiichuk2009, This `If you have DSC minted, you will not be able to redeem until you burn your DSC.` refers to users that have tight health factor. For any user that has a wide health factor, They don't need to burn DSC before redeeming some of their token, but no matter how wide your health factor is you can't redeem all your collateral except you burn your minted DSC as redeeming all your collateral without burning the DSC you have minted will always revert.

**Reply by nikita-biichuk** [reply]
@EngrPips, ah okay thanks, so basically doesn't user can redeem all collateral at once, but can redeem some amount but if their health factor stays ok?

**Reply by EngrPips** [reply]
> @EngrPips, ah okay thanks, so basically doesn't user can redeem all collateral at once, but can redeem some amount but if their health factor stays ok?

Correct.

**Reply by nikita-biichuk** [reply]
ok thanks!
